docs/manual/developer/06_contributing_with_content.md

```diff
@@ -1197,7 +1197,142 @@ See some of the existing testing code in `cmake/SSGCommon.cmake`.
+rules, this is why we turned remediations off in our scripts as mentioned
+earlier.
+
+For unit tests, we'll do something similar when setting up the
+testing container in our test scripts: We'll create a directory for
+the API resource object and persist it there.
+
+Note that the content uses the path `/kubernetes-api-resources` by
+default as a base for any API path for Kubernetes. So, be mindful of
+always using that.
+
+An example looks as follows:
+
+    kube_apipath="/kubernetes-api-resources"
+    machinev1beta1="/apis/machine.openshift.io/v1beta1"
+    machineset_apipath="$machinev1beta1/machinesets?limit=500"
+
+    # Create kubernetes resource directory path
+    mkdir -p "$kube_apipath/$machinev1beta1"
+
+    # Persist kubernetes resource
+    cat <<EOF > "$kube_apipath/$machineset_apipath"
+    {
+        "apiVersion": "v1",
+        ...
+    EOF
+
+When the scanner tests the content for this rule, it'll read the Kubernetes
+resource (both YAML or JSON are fine for this) and evaluate it accordingly.
+
+Note that some rules require a specific CPE to be set. e.g. there
+are some rules that are meant to run on a specific cloud platform.
+
+You'll see them from the `platforms` key of the rule:
+
+    platforms:
+    - ocp4-on-azure
+
+For rules like these, you'll also need to persist the appropriate CPE information (normally another Kubernetes object).
+
+The information needed varies on each CPE. These are defined
+in [`/shared/checks/oval`](https://github.com/ComplianceAsCode/content/tree/master/shared/checks/oval).
+
+For the `ocp4-on-azure` example, the CPE requires that the
+`infrastructures/cluster` resource specifies the platform
+as `Azure`. That would look as follows:
+
+    kube_apipath="/kubernetes-api-resources"
+
+    # Create infra file for CPE to pass
+    mkdir -p "$kube_apipath/apis/config.openshift.io/v1/infrastructures/"
+    cat <<EOF > "$kube_apipath/apis/config.openshift.io/v1/infrastructures/cluster"
+    {
+        "apiVersion": "config.openshift.io/v1",
+        "kind": "Infrastructure",
+        "metadata": {
+            "name": "cluster"
+        },
+        "spec": {
+            "platformSpec": {
+                "type": "Azure"
+            }
+        },
+        "status": {
+            "platform": "Azure",
+            "platformStatus": {
+                "azure": {
+                    "cloudName": "AzurePublicCloud"
+                },
+                "type": "Azure"
+            }
+        }
+    }
+    EOF
+
+Before running the unit tests, the content needs to be built to
+ensure that the latest changes are taken into account:
+
+    ./build_product ocp4
+
+Remember this needs to be done every time before running the unit tests.
+
+Now, with all of this set, we'll run the unit test!
+
+    tests/test_rule_in_container.sh \
+        --dontclean --logdir logs_bash \
+        --remediate-using bash \
+        --name ssg_test_suite \
+        --datastream build/ssg-ocp4-ds.xml \
+        <rule name>
```

Comment:
Can you double check this line, too, or give a complete example? I tried:
```
tests/test_rule_in_container.sh \
        --dontclean --logdir logs_bash \
        --remediate-using bash \
        --name ssg_test_suite \
        --datastream build/ssg-ocp4-ds.xml \
        route_ip_whitelist
```
and I'm getting:
```
INFO - The base image option has been specified, choosing Podman-based test environment.
INFO - Logging into logs_bash-3/test_suite.log
Traceback (most recent call last):
  File "/home/jhrozek/devel/compliance-as-code-content/tests/ssg_test_suite/test_env.py", line 485, in get_ip_address
    podman_output = subprocess.check_output(podman_cmd, stderr=subprocess.STDOUT)
  File "/usr/lib64/python3.9/subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib64/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['podman', 'inspect', 'time="2021-12-07T14:14:18+01:00" level=warning msg="Failed to decode the keys [\\"storage.options.ostree_repo\\"] from \\"/home/jhrozek/.config/containers/storage.conf\\"."\nti
me="2021-12-07T14:14:18+01:00" level=warning msg="Failed to decode the keys [\\"volume_path\\" \\"image_default_transport\\" \\"runtime\\" \\"conmon_path\\" \\"conmon_env_vars\\" \\"cgroup_manager\\" \\"init_path\\" \\"static_dir\\" \\"tmp
_dir\\" \\"max_log_size\\" \\"no_pivot_root\\" \\"cni_config_dir\\" \\"cni_plugin_dir\\" \\"infra_image\\" \\"infra_command\\" \\"enable_port_reservation\\" \\"label\\" \\"network_cmd_path\\" \\"num_locks\\" \\"EventsLogFilePath\\" \\"runt
imes\\" \\"runtimes.runc\\"] from \\"/home/jhrozek/.config/containers/containers.conf\\"."\ntime="2021-12-07T14:14:18+01:00" level=warning msg="Failed to decode the keys [\\"volume_path\\" \\"image_default_transport\\" \\"runtime\\" \\"con
mon_path\\" \\"conmon_env_vars\\" \\"cgroup_manager\\" \\"init_path\\" \\"static_dir\\" \\"tmp_dir\\" \\"max_log_size\\" \\"no_pivot_root\\" \\"cni_config_dir\\" \\"cni_plugin_dir\\" \\"infra_image\\" \\"infra_command\\" \\"enable_port_res
ervation\\" \\"label\\" \\"network_cmd_path\\" \\"num_locks\\" \\"EventsLogFilePath\\" \\"runtimes\\" \\"runtimes.runc\\"] from \\"/home/jhrozek/.config/containers/containers.conf\\"."\ntime="2021-12-07T14:14:18+01:00" level=warning msg="F
ailed to decode the keys [\\"volume_path\\" \\"image_default_transport\\" \\"runtime\\" \\"conmon_path\\" \\"conmon_env_vars\\" \\"cgroup_manager\\" \\"init_path\\" \\"static_dir\\" \\"tmp_dir\\" \\"max_log_size\\" \\"no_pivot_root\\" \\"c
ni_config_dir\\" \\"cni_plugin_dir\\" \\"infra_image\\" \\"infra_command\\" \\"enable_port_reservation\\" \\"label\\" \\"network_cmd_path\\" \\"num_locks\\" \\"EventsLogFilePath\\" \\"runtimes\\" \\"runtimes.runc\\"] from \\"/home/jhrozek/
.config/containers/containers.conf\\"."\ntime="2021-12-07T14:14:18+01:00" level=warning msg="Failed to decode the keys [\\"volume_path\\" \\"image_default_transport\\" \\"runtime\\" \\"conmon_path\\" \\"conmon_env_vars\\" \\"cgroup_manager
\\" \\"init_path\\" \\"static_dir\\" \\"tmp_dir\\" \\"max_log_size\\" \\"no_pivot_root\\" \\"cni_config_dir\\" \\"cni_plugin_dir\\" \\"infra_image\\" \\"infra_command\\" \\"enable_port_reservation\\" \\"label\\" \\"network_cmd_path\\" \\"n
um_locks\\" \\"EventsLogFilePath\\" \\"runtimes\\" \\"runtimes.runc\\"] from \\"/home/jhrozek/.config/containers/containers.conf\\"."\n3041947431caa6c7e643ac0f2cb08aed59cb997e84f6aed3311f8a87dd22ca36', '--format', '{{.NetworkSettings.IPAdd
ress}}']' returned non-zero exit status 125.
```

Reply:
Just tried it:
```
tests/test_rule_in_container.sh --dontclean --logdir logs_bash --remediate-using bash --name ssg_test_suite --datastream build/ssg-ocp4-ds.xml route_ip_whitelist
Setting console output to log level INFO
INFO - The base image option has been specified, choosing Podman-based test environment.
INFO - Logging into logs_bash/test_suite.log
ERROR - No tests found matching the rule ID(s) 'route_ip_whitelist'
WARNING - Nothing has been tested!
```